### DISCUSSION TITLE
Default view mode changed to "Show all branches"?

### DISCUSSION THREAD

**Discussion by jmvdkolk98** [gitextensions/gitextensions#9571] [Q&A] [answered]
It appears that there was a change in Git Extensions between versions 3.4.3 and 3.5.0 in what branches are shown. Can anyone tell me how to get the previous behavior back? Previously, Git Extensions would remember the setting for branches to show ("Show all branches" versus "Show current branch only" between sessions. As of version 3.5.0, my Git Extensions always defaults to "Show all branches". I have gone back to version 3.4.3 in hopes that the previous behavior can be restored through some setting. I haven't been able to find an option to restore the previous behavior myself.

Also, the order of branches in the left pane has changed. In version 3.4.3, my current branch is always the topmost branch, and branches seem to be in order of activity. With version 3.5.0 this order has changed, and my current branch is now somewhere in the middle of the (sometimes long) list. Is there a setting to restore the previous behavior? I haven't been able to find such a setting, unfortunately.

**Comment by pmiossec** [comment, marked as answer]
> It appears that there was a change in Git Extensions between versions 3.4.3 and 3.5.0 in what branches are shown. Can anyone tell me how to get the previous behavior back? Previously, Git Extensions would remember the setting for branches to show ("Show all branches" versus "Show current branch only" between sessions. As of version 3.5.0, my Git Extensions always defaults to "Show all branches". I have gone back to version 3.4.3 in hopes that the previous behavior can be restored through some setting. I haven't been able to find an option to restore the previous behavior myself.

Regression that should be fixed by #9551 waiting to be released. 

> 
> Also, the order of branches in the left pane has changed. In version 3.4.3, my current branch is always the topmost branch, and branches seem to be in order of activity. With version 3.5.0 this order has changed, and my current branch is now somewhere in the middle of the (sometimes long) list. Is there a setting to restore the previous behavior? I haven't been able to find such a setting, unfortunately.

The setting is in "appearance" => "sort branches by "

**Reply by jmvdkolk98** [reply]
Thank you @pmiossec. I'm sorry I didn't find the previous mention of this issue. I will be anxiously awaiting the release of version 3.5.4.

**Reply by RussKie** [reply]
![image](https://user-images.githubusercontent.com/4403806/133277378-62830615-4e64-4309-a452-95d31da48333.png)

**Reply by jmvdkolk98** [reply]
Thank you @RussKie . Is "Author date" really the previous behavior? I just upgraded to see these options and will have to go back, but I thought that the current branch was always at the top, wasn't it? I frequently switch to "old" branches, and do not recall them ever being anywhere else in the sort order. Just asking.

**Reply by RussKie** [reply]
To be honest I don't remember all related discussions, it was that long since this was implemented. We had some sorting mechanism in late 2.5x or early 3.0 but then it got broken for few releases... Eventually it was implemented as it is now in https://github.com/gitextensions/gitextensions/pull/8427.
If you trawl through the PR and related issues you can probably find the answer to your questions...

**Reply by RussKie** [reply]
As far as the screenshots goes - this is how I have it configured. You can choose your own default.

**Reply by jmvdkolk98** [reply]
Thank you @RussKie for the feedback. I'll work with what is available.
